### PR TITLE
upgrade `sqlalchemy` base version to 2.0.8

### PR DESCRIPTION
**Pull request by wssheldon**
https://github.com/kvesteri/sqlalchemy-utils/issues/505 Appears to be resolved. I'd like to use the `mypy` types that were introduced in `2.0`

[0] - https://docs.sqlalchemy.org/en/20/orm/extensions/mypy.html

### CODE REVIEW

**Comment by mvilanova** [comment]
Sounds good. Let's upgrade and test for any other breaking changes.

**Review by mvilanova — APPROVED** [review]

**Comment by wssheldon** [comment]
> Sounds good. Let's upgrade and test for any other breaking changes.

I'm working through some code changes right now to get the tests to pass. Will wait to merge till those are resolved.

File: tests/signal/test_signal_service.py [hunk 1]
```diff
@@ -67,31 +66,26 @@ def test_filter_actions_deduplicate(session, signal, project):
...
     session.commit()
 
     # create deduplicate signal filter
     signal_filter = SignalFilter(
         name="test",
         description="test",
-        expression=[
-            {"or": [{"model": "EntityType", "field": "id", "op": "==", "value": entity_type.id}]}
-        ],
+        expression=[{"or": [{"model": "Entity", "field": "id", "op": "==", "value": entity.id}]}],
```
Comment (wssheldon) [inline tests/signal/test_signal_service.py:88]: Changed to resolve [test failure](https://github.com/Netflix/dispatch/actions/runs/4624040233/jobs/8178465630?pr=3228)

This passes.

cc @kevgliss

File: src/dispatch/database/manage.py [hunk 2]
```diff
@@ -133,20 +134,20 @@ def init_database(engine):
...
+    with engine.begin() as connection:
+        connection.execute(CreateSchema(schema_name, if_not_exists=True))
 
     # set the schema for table creation
     tables = get_tenant_tables()
 
     schema_engine = engine.execution_options(
         schema_translate_map={
             None: schema_name,
+            "dispatch_core": "dispatch_core",
```
Comment (wssheldon) [inline src/dispatch/database/manage.py:150]: Added to resolve [test failure](https://github.com/Netflix/dispatch/actions/runs/4624040233/jobs/8178465630?pr=3228)

This passes.

cc @kevgliss  @mvilanova

Comment (wssheldon) [reply src/dispatch/database/manage.py:150]: This one seems wrong.

**Comment by wssheldon** [comment]
Looks like the `sqlalchemy_filters` project has gotten new maintainers and support for `1.4` was released. Which will revert a lot of the forked code added in this PR.

https://github.com/Netflix/dispatch/pull/3283

**Pull request closed without merging**